test/runtests.jl

```diff
@@ -1469,3 +1469,16 @@ end
     @test cor(Fill(3, 4, 5)) ≈ cor(fill(3, 4, 5)) nans=true
     @test cor(Fill(3, 4, 5), dims=2) ≈ cor(fill(3, 4, 5), dims=2) nans=true
 end
+
+@testset "Structured broadcast" begin
+    D = Diagonal(1:5)
+    @test D + Zeros(5,5) isa Diagonal
+    @test D - Zeros(5,5) isa Diagonal
```

Comment:
These tests should test `D .+ Zeros(5,5)` and so on, as currently the `+/-` methods don't use broadcasting and won't use `fzeros`